### ISSUE TITLE
allow setting `aria-current` on `menuitem` with `href`

### ISSUE THREAD

**Issue by stefanprobst** [adobe/react-spectrum#7587]
### Provide a general summary of the issue here

menu items can have a href, but currently don't allow setting `aria-current`.

the apg does mention "aria-current" for "menuitem": https://www.w3.org/WAI/ARIA/apg/patterns/menubar/examples/menubar-navigation/

### 🤔 Expected Behavior?

`aria-current` should be forwarded to the `<a role="menuitem" href="/" aria-current="page">`

### 😯 Current Behavior

`aria-current` is not forwarded to the `<a role="menuitem" href="/">`

### 🖥️ Steps to Reproduce

https://codesandbox.io/p/sandbox/awesome-brown-krr5tq?file=%2Fsrc%2FApp.js%3A11%2C32

### Version

latest

### What browsers are you seeing the problem on?

Firefox

### What operating system are you using?

linux

**Comment by snowystinger** [comment]
Instead of aria-current, you should use selection in a Menu https://codesandbox.io/p/sandbox/modest-violet-38kpgg

**Comment by stefanprobst** [comment]
is that because `<a href="/" role="menuitem">` is announced as a menuitem rather than as a link?

**Comment by snowystinger** [comment]
right, explicit role wins over implicit
